interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1047 @@
+later be searched easily outside of Cantera.
+
+Setting overwrite_files to True will allow yaml2ck to delete a prior version of a
+mechanism and write a file with the same name. If it is False then yaml2ck will throw
+an error.
+
+The cantera-validate keyword is a boolean choice to test that the converted 
+Chemkin-formatted mechanism will load back into Cantera.
+
+Supported Thermodynamics: 
+    NASA7
+
+Supported Reaction models: 
+    Elementary, Three-Body, Falloff, Pressure-Dependent-Arrhenius (PLOGs), Chebyshev
+
+Unsupported Thermodynamic Property Models:
+    NASA9, Shomate, constant-Cp
+
+Unsupported Reaction models (not comprehensive): 
+    Chemically-activated, Blowers-Masel, Surface Reactions
+"""
+
+import os
+import sys
+import math
+import re
+import argparse
+from pathlib import PurePath, Path
+from textwrap import fill
+import cantera as ct
+
+# number of calories in 1000 Joules
+CALORIES_CONSTANT = 4184.0
+
+# Conversion from 1 debye to coulomb-meters
+DEBYE_CONVERSION = 3.33564095e-30
+
+
+def Eformat(val, precision=7, exp_digits=3):
+    """
+    Formats numbers in a pretty format for printing
+
+    :param val:
+        Float to be formatted into a str
+    :param precision:
+        How many digits after the decimal
+    :param exp_digits:
+        How many digits the exponent has
+    """
+    if math.isinf(val) and val > 0:
+        return " inf"
+
+    elif math.isinf(val):
+        return "-inf"
+
+    # round to correct number of significant digits
+    if val != 0:
+        val = round(val, precision - int(math.floor(math.log10(abs(val)))) - 1)
+
+    s = f"{val: .{precision}e}"
+
+    mantissa, exponent = s.split("e")
+    exp_digits += 1  # +1 due to sign
+
+    return f"{mantissa}E{int(exponent):+0{exp_digits}d}"
+
+
+def build_header_text(solution, max_width=120):
+    """
+    Creates the header text for all output files
+
+    :param solution:
+        The `solution` object being converted
+    :param max_width:
+        The maximum width of lines before they start to wrap
+    """
+    text = []
+
+    if "description" in solution.input_header:
+        text_wrap = lambda t: fill(
+            t,
+            width=max_width,
+            break_long_words=False,
+            break_on_hyphens=False,
+            initial_indent="! ",
+            subsequent_indent="!   ",
+        )
+
+        text = solution.input_header["description"].splitlines()
+        text = [text_wrap(line) for line in text]
+    else:
+        note = ""
+
+    ct_note = (
+        f"! Chemkin file was converted from a Cantera {ct.__version__} "\
+        "`solution` object"
+    )
+
+    has_ct_note = False
+    for n, line in enumerate(text):
+        if "! Chemkin file was converted from a Cantera" in line:
+            text[n] = ct_note
+            has_ct_note = True
+            break
+
+    if not has_ct_note:
+        text.append(f"!\n{ct_note}\n! \n\n")
```

Comment:
What's the purpose of this check?

Reply:
Purpose is to check if the boilerplate "This was converted by Cantera version blah blah" has already been written once (back and forth conversions). If it has been written, overwrite it with the new boilerplate (to update the version). If it has not been written then append it to the existing text.

Reply:
I'm concerned that we risk losing data in the case of overwriting.

That said, I can see how the write-read-write cycle could lead to an explosion of data here. Maybe we should add an option `--no-header` to not write anything other than the mechanism out? I think that should cover the most common cases... the reason to write out the Cantera version is to help with debugging, but if you add the `--no-header` option then at least you have to know a little bit what you're doing.

Reply:
I modified the check to:

```
is_ct_header = ("! Chemkin file was converted from a Cantera" == line[:43]
            and "Solution object" == line[-17:])
if is_ct_header:
```

So there will still be overwriting but the only way someone would have a line overwritten is if they wrote out exactly this and decided to put information in the version number spot. Is strengthening the check a good compromise?